things/service.go

```diff
@@ -532,6 +541,71 @@ func (svc service) ListClientsByGroup(ctx context.Context, token, groupID string
 	}, nil
 }
 
+func (svc service) VerifyConnectionsHttp(ctx context.Context, token string, thingIDs, groupIDs []string) (mgclients.ConnectionsPage, error) {
+	res, err := svc.identify(ctx, token)
+	if err != nil {
+		return mgclients.ConnectionsPage{}, err
+	}
+
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(thingIDs)+len(groupIDs))
+
+	for _, thID := range thingIDs {
+		wg.Add(1)
+		go func(thingID string) {
+			defer wg.Done()
+			_, err := svc.authorize(ctx, res.GetDomainId(), auth.UserType, auth.UsersKind, res.GetId(), auth.ViewPermission, auth.ThingType, thingID)
+			if err != nil {
+				errChan <- err
+			}
+		}(thID)
```

Comment:
Instead of go routines , we can try to use https://pkg.go.dev/golang.org/x/sync/errgroup.
like we use ing `cmd/<svc>/main.go`